crates/wasi-nn/src/ctx.rs

```diff
@@ -38,15 +47,29 @@ impl Ctx {
 /// This struct solely wraps [Ctx] in a `RefCell`.
 pub struct WasiNnCtx {
     pub(crate) ctx: RefCell<Ctx>,
+    pub map_dir: Option<Vec<(String, String)>>,
 }
 
 impl WasiNnCtx {
     /// Make a new `WasiNnCtx` with the default settings.
     pub fn new() -> WasiNnResult<Self> {
         Ok(Self {
             ctx: RefCell::new(Ctx::new()?),
+            map_dir: None,
         })
     }
+
+    // Save mapped paths so they can be used by backends later.
+    pub fn set_paths(&mut self, paths: &Vec<(String, String)>) {
+        if paths.len() > 0 {
+            self.map_dir = Some(Vec::<(String, String)>::new());
+            let md = self.map_dir.as_mut().unwrap();
+
+            for (guest, host) in paths.iter() {
+                md.push((guest.clone(), host.clone()));
+            }
+        }
```

Comment:
```suggestion
        self.map_dir = paths.clone();
```

I think it should just work to clone the entire vector. And we might as well just switch to using `Vec` as the struct type since one can detect the presence of mapped directories with `is_empty()`; why re-wrap in an `Option`?